### DISCUSSION TITLE
Request for Guidance on Radiative Transfer Data Simulation Using PyARTS 2.4

### DISCUSSION THREAD

**Discussion by curious777x** [atmtools/arts#888] [Q&A]
Hello PyARTS Community,

I hope you're all doing well! I am currently working with PyARTS 2.4 due to compatibility issues with newer versions. I have successfully followed the user guide to create basic workspaces, define variables, load XML files, and understand some core functionalities of the software. However, I'm facing difficulties when it comes to simulating radiative transfer data for the Martian atmosphere.

Although I have a general understanding of PyARTS, I am unsure of the precise steps or tools required to simulate radiative transfer for a Martian environment, including handling atmospheric composition, temperature profiles, and radiative transfer processes specific to Mars.

Could someone kindly guide me or point me towards relevant resources, examples, or documentation that would help me with the following?

Setting up a Martian atmosphere in PyARTS 2.4: What are the key parameters (e.g., pressure, temperature, gases) that need to be defined?

Simulating radiative transfer: Which modules or functions should I use to run the actual radiative transfer simulations for Mars?

Dealing with Mars-specific conditions: Are there any special considerations for Mars (e.g., dust, CO2 concentration, surface reflectivity) that I need to take into account when performing radiative transfer simulations?

The main issue i am facing is in the simulation of radiative transfer for martian atmosphere as many things are not properly clear in user guide.

Any help, examples, or advice from the community would be greatly appreciated!

Thank you in advance for your support, and I look forward to any insights you can provide.

**Comment by curious777x** [comment]
Can anyone please help

**Comment by riclarsson** [comment]
Dear @curious777x ,

It really sounds like you are starting a new project altogheter.  Is this true?  If so, please use a more modern version of ARTS.  I don't think there's anyone in the ARTS development team using such an ancient version of ARTS anymore ;) so I do not think you will get answers here.

Otherwise, can you attach the piece of code that is not doing what you expect it to do?

**Reply by curious777x** [reply]
Hello,

Thank you, @riclarsson , for your previous reply!

1) Is there a .whl file available for the latest PyARTS version (e.g., 2.6.12)? I’m working in an offline setup and can only transfer files from an internet-connected system. I successfully installed PyARTS 2.4 using the .whl on PyPI but couldn’t install newer versions despite several attempts.

2) I’m working on simulating radiative transfer for the Martian atmosphere but haven’t found practical guidance in the user manual. Can anyone provide steps or examples for simulating Martian Atmosphere?

Any help would be greatly appreciated!

Thank you!

**Reply by riclarsson** [reply]
If you have no access to the internet we cannot help you.  The .whl wouldn't help you without internet anyways because of dependencies.  I recommend you build it manually with approved packages, but I cannot tell you how to install the ARTS dependencies anyways without internet...  But the CMake system should help inidcate what packages you need.  Good luck!

About Mars, just use the normal atmospheric setup available in arts-xml-data for Mars.  Your line-by-line data must be adapted to the CO2 atmosphere, though, and doing that really depends on what you want to measure.  Perhaps a simple mass-scaling of the broadening parameter is enough.

**Reply by curious777x** [reply]
Hello,

Thank you, @riclarsson, for your previous reply!

I am currently using PyARTS 2.4.0 in an offline setup for radiative transfer simulations in the Martian atmosphere. I successfully converted the demo file DemoMars_1D_clearsky.arts into Python using convert_arts.py. However, when executing the converted script, I encounter the following issue:

A parsing error occurs in the file DemoMarsAtmo1D.arts during the execution of DemoMars_1D_clearsky.py, with the following message:
"Exception during the execution of Agenda:
Run-time error in Method: AtmFieldsCalc
Method AtmFieldsCalc needs input variable: nlte_vibrational_energies"

To move forward, I would greatly appreciate any guidance on addressing this parsing error while running the demo file. Any solutions or advice for resolving this issue would be highly helpful.

Thank you for your time and support!

[DemoMars_fullRT_1D_clearsky.py.txt](https://github.com/user-attachments/files/18176242/DemoMars_fullRT_1D_clearsky.py.txt)
[DemoMarsAtmo1D.arts.txt](https://github.com/user-attachments/files/18176285/DemoMarsAtmo1D.arts.txt)
[Screenshot 2024-12-17 225935](https://github.com/user-attachments/assets/166e2cd1-9868-4877-91ed-08ff7497fac2)

**Comment by curious777x** [comment]
Can anyone please help
